hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveClientPool.java

```diff
@@ -110,57 +103,4 @@ public void testGetTablesFailsForNonReconnectableException() throws Exception {
     AssertHelpers.assertThrows("Should throw exception", MetaException.class,
             "Another meta exception", () -> clients.run(client -> client.getTables("default", "t")));
   }
-
-  @Test
-  public void testConnectionFailureRestoreForMetaException() throws Exception {
```

Comment:
These tests seem valuable if the client doesn't retry on its own. What about passing in an option to turn on retries and then using that to test retries here?

Reply:
Good idea, done.  Put the test back.

Side note: the flexibility of making it configurable is nice, and is a potential soution the issue I mentioned, I am tempted to set it to true in HiveClientOperations::acquireLock in this line to address the issue I mentioned above.

```
LockResponse lockResponse = metaClients.run(client -> client.lock(lockRequest));
```

I can see from Hive point of view why they don't want to auto-retry, but most exception scenarios are not going to be in the corner case of having succeeded in the backend and leaving the lock.  In all honesty, if the Iceberg job fails because lock() throws TTransportException, end user will just try again but it is a wasted effort.  Not sure if any thoughts on it? 
 cc @pvary

Reply:
Having both Iceberg and Hive retry could also be problematic / surprising. We will just do NxM retries (2x2 in the default case).

I see 2 typical error scenario in case of `lock`:

1. Can not reach the HMS - connection is broken, or something like this - Retry is a perfect solution
2. The locks are queued, but we have to wait and something happens in the meantime - Retry will be blocked forever (or until the lock timeout is reached for the previously queued locks) - Retry is problematic / unnecessary.

How typical is that the first request on the connection is the `lock`? If it happens often then the 1st situation could happen often. If we usually get the metadata for the table and then try to lock then the 1st situation is unlikely to happen.

I would prefer to keep the retry login in one place (`RetryingMetaStoreClient`) and simplify the Iceberg code by removing the retry altogether, but I can accept keeping the code as it is - we most probably will turn off the Iceberg retries in our deployment.

Reply:
@pvary yea this PR disables the retry for Hive case, but leaves an option for turning it on to a particular caller, to address your concern.  

About lock(), thanks for the reply, maybe you are right that it is not typical case to have TTransportException especially if getMetadata has already succeeded on same connection.  

The only thing bothering me is that there is no way for the client to distinguish case 1 and 2 based on exception type, and thus nothing they can do really except retry.  There is a ShowLock call, and potentially they can check if there is a queued lock from the failed call, is it it is their own , and if so then unlock it.  But I'm not sure how reliable this is.  Anyway its maybe for another review then.

Reply:
Maybe we could make Hive lock() SafeToRetry?
I have to think this through, but if we release the enqueued lock in case of unexpected error inside implementation, and also delete the previous version of the lock if someone already created a lock for this transaction.